platform/kindle/koreader.sh

```diff
@@ -232,20 +236,20 @@ if [ "${STOP_FRAMEWORK}" == "no" -a "${INIT_TYPE}" == "upstart" ] ; then
 fi
 
 # stop cvm (sysv & framework up only)
-if [ "${STOP_FRAMEWORK}" == "no" -a "${INIT_TYPE}" == "sysv" ] ; then
+if [ "${STOP_FRAMEWORK}" = "no" ] && [ "${INIT_TYPE}" = "sysv" ] ; then
 	logmsg "Stopping cvm . . ."
 	killall -stop cvm
 fi
 
 # finally call reader
 logmsg "Starting KOReader . . ."
 # That's not necessary when using KPVBooklet ;).
-if [ "${FROM_KUAL}" == "yes" ] ; then
+if [ "${FROM_KUAL}" = "yes" ] ; then
 	eips_print_bottom_centered "Starting KOReader . . ." 1
 fi
 
 # we keep maximum 500K worth of crash log
-cat crash.log 2> /dev/null | tail -c 500000 > crash.log.new
+< crash.log tail -c 500000 > crash.log.new
```

Comment:
This will print out a misleading error message if crash.log does not exist. How about we change it to the following?

```bash
if [ -e crash.log ]; then
    tail -c 500000 crash.log > crash.log.new
    mv -f crash.log.new crash.log
fi
```